py-rse/bash-basics.Rmd

```diff
@@ -1743,54 +1842,55 @@ though,
 the multi-line format is much easier to read—compare this:
 
 ```shell
-$ for filename in *_temp.csv
+$ for filename in s*.txt
 > do
->   echo $filename
->   head -n 4 $filename | tail -n 3
+> echo $filename
+> head -n 19 $filename | tail -n 11
 > done
+
 ```
 
 with this:
 
 ```shell
-$ for filename in *_temp.csv; do echo $filename; head -n 4 $filename | tail -n 3; done
+$ for filename in s*.txt; do echo $filename; head -n 19 $filename | tail -n 11; done
 ```
 
 ## How can I create new filenames automatically? {#py-rse-bash-basics-autoname}
 
-Suppose we want to create a backup copy of each precipitation data file.
+Suppose we want to create a backup copy of each book whose name starts with m.
 If we don't want to change the files' names,
 we can do this with `cp`:
 
 ```shell
-$ cd ~/climate-data
+$ cd ~/zipfs-law
 $ mkdir backup
-$ cp cleaned/*_prec.csv backup
+$ cp data/m*.txt backup
 ```
 
-But what if we want the backup files to have the [extension][filename-extension] `.bak` instead of `.csv`?
+But what if we want the backup files to have the [extension][filename-extension] `.bak` instead of `.txt`?
 `cp` can do this for a single file:
 
 ```shell
-$ cp cleaned/andamooka_prec.csv backup/andamooka_prec.bak
+$ cp data/moby_dick.txt backup/moby_dick.txt
 ```
 
 but can't do all the files at once:
 
 ```shell
-$ cp cleaned/*_prec.csv backup/*_prec.bak
+$ cp data/m*.txt backup/m*.bak
 ```
 
 ```text
 usage: cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file target_file
        cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file ... target_directory
 ```
 
-`backup/*_prec.bak` doesn't match anything—those files don't yet exist—so
+`backup/*.bak` doesn't match anything—those files don't yet exist—so
 what we are actually asking `cp` to do is:
 
 ```shell
-$ cp cleaned/andamooka_prec.csv cleaned/badingarra_prec.csv cleaned/bellambi_prec.csv cleaned/tuggeranong_prec.csv
+$ cp moby_dick.txt mysterious_affair_at_styles.txt
```

Comment:
This command would work - it would simply replace the contents of `mysterious_affair_at_styles.txt` with those in `moby_dick.txt`. I think this section was originally written for a directory with more than two files starting with m. There aren't more than two books starting with any letter, so this section might need a minor re-think.